fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/mv/MaterializedViewRewriter.java

```diff
@@ -68,11 +70,44 @@ public OptExpression visitLogicalProject(OptExpression optExpression, Materializ
 
         LogicalProjectOperator projectOperator = (LogicalProjectOperator) optExpression.getOp();
 
+        Map<ColumnRefOperator, ScalarOperator> replaceMap = new HashMap<>();
+        replaceMap.put(context.queryColumnRef, context.mvColumnRef);
+        ReplaceColumnRefRewriter replaceColumnRefRewriter = new ReplaceColumnRefRewriter(replaceMap);
+
         Map<ColumnRefOperator, ScalarOperator> newProjectMap = Maps.newHashMap();
         for (Map.Entry<ColumnRefOperator, ScalarOperator> kv : projectOperator.getColumnRefMap().entrySet()) {
             if (kv.getValue().getUsedColumns().contains(context.queryColumnRef)) {
                 if (kv.getValue() instanceof ColumnRefOperator) {
                     newProjectMap.put(context.mvColumnRef, context.mvColumnRef);
+                } else if (kv.getValue() instanceof CallOperator && kv.getValue().equals(context.conditionCall)) {
+                    Type[] newArgsType = new Type[kv.getValue().getChildren().size()];
+                    if (kv.getValue() instanceof CaseWhenOperator) {
+                        for (int i = 0; i < kv.getValue().getChildren().size(); i++) {
```

Comment:
what's the case? how about extracting it to a separated method and add some comments ?